src/main/java/core/mate/academy/service/ExcavatorProducer.java

```diff
@@ -0,0 +1,16 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Excavator;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExcavatorProducer implements MachineProducer<Excavator> {
```

Comment:
The `MachineProducer` interface should be parameterized in such a way that it cannot be implemented with a type that is not a subtype of `Machine`. This implementation is correct as `Excavator` is a subtype of `Machine`. Good job following the checklist here.